pkg/operator/workloads/api_workload.go

```diff
@@ -38,8 +37,159 @@ import (
 const (
 	apiContainerName       = "api"
 	tfServingContainerName = "serve"
+
+	defaultPortInt32, defaultPortStr     = int32(8888), "8888"
+	tfServingPortInt32, tfServingPortStr = int32(9000), "9000"
 )
 
+type APIWorkload struct {
+	BaseWorkload
+}
+
+func populateAPIWorkloadIDs(ctx *context.Context, latestResourceWorkloadIDs map[string]string) {
+	for _, api := range ctx.APIs {
+		if api.WorkloadID != "" {
+			continue
+		}
+		if workloadID := latestResourceWorkloadIDs[api.ID]; workloadID != "" {
+			api.WorkloadID = workloadID
+			continue
+		}
+		api.WorkloadID = generateWorkloadID()
+	}
+}
+
+func extractAPIWorkloads(ctx *context.Context) []Workload {
+	workloads := make([]Workload, 0, len(ctx.APIs))
+
+	for _, api := range ctx.APIs {
+		workloads = append(workloads, &APIWorkload{
+			singleBaseWorkload(api, ctx.App.Name, workloadTypeAPI),
+		})
+	}
+
+	return workloads
+}
+
+func (aw *APIWorkload) Start(ctx *context.Context) error {
+	api := ctx.APIs.OneByID(aw.GetSingleResourceID())
+
+	k8sDeloymentName := internalAPIName(api.Name, ctx.App.Name)
+	k8sDeloyment, err := config.Kubernetes.GetDeployment(k8sDeloymentName)
+	if err != nil {
+		return err
+	}
+	hpa, err := config.Kubernetes.GetHPA(k8sDeloymentName)
+	if err != nil {
+		return err
+	}
+
+	desiredReplicas := getRequestedReplicasFromDeployment(api, k8sDeloyment, hpa)
+
+	var deploymentSpec *kapps.Deployment
+
+	switch api.ModelFormat {
+	case userconfig.TensorFlowModelFormat:
+		deploymentSpec = tfAPISpec(ctx, api, aw.WorkloadID, desiredReplicas)
+	case userconfig.ONNXModelFormat:
+		deploymentSpec = onnxAPISpec(ctx, api, aw.WorkloadID, desiredReplicas)
+	default:
+		return errors.New(api.Name, "unknown model format encountered") // unexpected
+	}
+
+	_, err = config.Kubernetes.ApplyIngress(ingressSpec(ctx, api))
+	if err != nil {
+		return err
+	}
+
+	_, err = config.Kubernetes.ApplyService(serviceSpec(ctx, api))
+	if err != nil {
+		return err
+	}
+
+	_, err = config.Kubernetes.ApplyDeployment(deploymentSpec)
+	if err != nil {
+		return err
+	}
+
+	_, err = config.Kubernetes.ApplyHPA(hpaSpec(ctx, api))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (aw *APIWorkload) IsSucceeded(ctx *context.Context) (bool, error) {
```

Comment:
would be nice to add unit test to this somehow...not sure

Reply:
Agreed, I'll add a ticket